src/sidebar/components/test/tag-editor-test.js

```diff
@@ -268,49 +269,90 @@ describe('TagEditor', function () {
-      wrapper.find('input').instance().value = '  ';
-      selectOptionViaEnter(wrapper);
-      assertAddTagsFail();
-    });
+    context('When using the "Enter" key', () => {
+      it('should not add a tag if the <input> is empty', () => {
+        const wrapper = createComponent();
+        wrapper.find('input').instance().value = '';
+        selectOptionViaEnter(wrapper);
+        assertAddTagsFail();
+      });
 
-    it('should not add a tag if its a duplicate of one already in the list', () => {
-      const wrapper = createComponent();
-      wrapper.find('input').instance().value = 'tag1';
-      selectOptionViaEnter(wrapper);
-      assertAddTagsFail();
-    });
+      it('should not add a tag if the <input> value is only white space', () => {
+        const wrapper = createComponent();
+        wrapper.find('input').instance().value = '  ';
+        selectOptionViaEnter(wrapper);
+        assertAddTagsFail();
+      });
 
-    it('should not add a tag when pressing "Tab" and there are no suggestions', () => {
-      const wrapper = createComponent();
-      fakeTagsService.filter.returns([]);
-      wrapper.find('input').instance().value = 'tag33';
-      typeInput(wrapper);
-      selectOptionViaTab(wrapper);
-      assertAddTagsFail();
+      it('should not add a tag if its a duplicate of one already in the list', () => {
+        const wrapper = createComponent();
+        wrapper.find('input').instance().value = 'tag1';
+        selectOptionViaEnter(wrapper);
+        assertAddTagsFail();
+      });
     });
 
-    it('should not a tag when pressing "Tab" and no suggestions are found', () => {
-      const wrapper = createComponent();
-      wrapper.find('input').instance().value = 'tag3';
-      // note: typeInput() opens the suggestions list
-      selectOptionViaTab(wrapper);
-      assertAddTagsFail();
+    context('Using the "Tab" key', () => {
+      it('should add the tag as typed when there are no suggestions', () => {
+        const wrapper = createComponent();
+        fakeTagsService.filter.returns([]);
+        wrapper.find('input').instance().value = 'tag33';
+        typeInput(wrapper);
+        selectOptionViaTab(wrapper);
+        assertAddTagsSuccess(wrapper, ['tag1', 'tag2', 'tag33']);
+        // ensure focus is still on the input field
+        assert.equal(document.activeElement.nodeName, 'INPUT');
+      });
+
+      it('should add the tag as typed when there are multiple suggestions', () => {
+        const wrapper = createComponent();
+        fakeTagsService.filter.returns([]);
+        wrapper.find('input').instance().value = 't';
+        typeInput(wrapper);
+        selectOptionViaTab(wrapper);
+        assertAddTagsSuccess(wrapper, ['tag1', 'tag2', 't']);
+        // ensure focus is still on the input field
+        assert.equal(document.activeElement.nodeName, 'INPUT');
+      });
+
+      it('should add the suggested tag when there is exactly one suggestion', () => {
```

Comment:
So, I'm sure this is by design, but I will just confirm.

Case 1:
type "moo" and...
If I have exactly 1 tag suggestion of "zoo moo" 
type `Enter` or `,`
A "moo" tag appears

Case 2:
type "moo" and...
If I have exactly 1 tag suggestion of "zoo moo" 
type `Tab`
A "zoo moo" tag appears

I think that's what we want, but I want to make 100% sure the difference between `Enter` or `,` and `Tab` is clear in this case.